### PR TITLE
docs(grid): updated with API docs and example

### LINKED ISSUE #3108: docs(2x-grid): update with breakpoints api
### Summary

The website needs to show how the `breakpoint` API works. This is mostly a content design thing, but also UX and dev need to be in the room.

### Justification

Closes https://github.com/carbon-design-system/carbon/issues/10687

### Desired UX and success metrics

An Carbon maker should be able to visit the 2x Grid page and leave with a full comprehension of the breakpoint API.

### Must-have functionality

Using our `breakpoint` [mixins](https://github.com/carbon-design-system/carbon/blob/ff7418e115708635cf4936b6e7f999824af1bc57/packages/layout/scss/_breakpoint.scss#L134-L243) to add styles (hide or show components) for different breakpoints. Something like:

```css

.component {
    display: inline;

    /* applies styles inside mixin on md breakpoint or below, like max-width */
    @include carbon--breakpoint-down('md') {
        display: none; 
    }
}
```

the breakpoint mixins include:
- `carbon--breakpoint-down()`
- ` carbon--breakpoint-up()` || `breakpoint()`
- `carbon--breakpoint-between()` this takes 2 args, to apply styles in between 2 breakpoints i.e. md - lg
- `carbon--largest-breakpoint()`

### PR DESCRIPTION
**Pull request by sstrubberg**
Closes #3108 

Closes [10687](https://github.com/carbon-design-system/carbon/issues/10687)

#### New

Added progressive disclosure to the API docs from https://github.com/carbon-design-system/carbon/blob/main/packages/grid/docs/sass.md#api as well as the Helpers in Storybook.

### CODE REVIEW

**Comment by aagonzales** [comment]
@sstrubberg Is the code block an example of the API? Adding a little description text under the API heading might help users understand this section better.

File: src/pages/guidelines/2x-grid/implementation.mdx [hunk 1]
```diff
@@ -1343,6 +1344,42 @@ Image size can change ratios when the breakpoint changes.
...
+about our API.
+
+```css
+.component {
+  display: inline;
+
+  /* applies styles inside mixin on md breakpoint or below, like max-width */
+  @include carbon--breakpoint-down('md') {
+    display: none;
+  }
```
Comment (tay1orjones) [inline src/pages/guidelines/2x-grid/implementation.mdx:1361]: The adding of helper classes in https://github.com/carbon-design-system/carbon/pull/11786 was intended to replace the need for this type of manual config. Should we just point them to those classes instead?

File: src/pages/guidelines/2x-grid/implementation.mdx [hunk 2]
```diff
@@ -1343,6 +1344,42 @@ Image size can change ratios when the breakpoint changes.
...
+| `breakpoint-prev`        | Get the value of the previous breakpoint                                            | `@function` |               |
+| `is-smallest-breakpoint` | Check to see if the given breakpoint is the smallest breakpoint                     | `@function` |               |
+| `is-largest-breakpoint`  | Check to see if the given breakpoint is the largest breakpoint                      | `@function` |               |
+| `breakpoint-up`          | Generate a media query from the width of the given breakpoint to infinity           | `@mixin`    |               |
+| `breakpoint-down`        | Generate a media query that applies below the maximum width of the given breakpoint | `@mixin`    |               |
+| `breakpoint-between`     | Generate a media query for the range between the lower and upper breakpoints        | `@mixin`    |               |
+| `largest-breakpoint`     | Generate a media query for the largest breakpoint                                   | `@mixin`    |               |
+| `breakpoint`             | Generate a media query for a given breakpoint, alias of `breakpoint-up`             | `@mixin`    |               |
```
Comment (tay1orjones) [inline src/pages/guidelines/2x-grid/implementation.mdx:1381]: Could we instead link to the grid package Sass API docs? 

https://github.com/carbon-design-system/carbon/blob/main/packages/grid/docs/sass.md#api

If we duplicate this here, one of them is bound to become out of date. I don't think we surface full Sass API information anywhere else on the website.

File: src/pages/guidelines/2x-grid/implementation.mdx (hunk 1 above)
Comment (jnm2377) [reply src/pages/guidelines/2x-grid/implementation.mdx:1361]: I think the grid mixins would still be needed for other use cases, no? Like if you wanted to change the layout or styles of something at mobile breakpoints (not necessarily just hide it). @tay1orjones but we should also add those classes and surface them here! I think surfacing all options is good.

File: src/pages/guidelines/2x-grid/implementation.mdx (hunk 2 above)
Comment (jnm2377) [reply src/pages/guidelines/2x-grid/implementation.mdx:1381]: I think we do have a resource tile that links to the package readme. I think there's a big need to surface these here as well. We tend to do this with all our elements stuff like spacing, motion sass - https://carbondesignsystem-fq9t0896o-carbon-design-system.vercel.app/guidelines/motion/code but we don't do anything for grid. We don't surface that code anywhere on our website. Totally get the issue about these becoming out of date though. I'm not sure of a workaround for that. How often do we make changes to the grid API?

File: src/pages/guidelines/2x-grid/implementation.mdx (hunk 1 above)
Comment (sstrubberg) [reply src/pages/guidelines/2x-grid/implementation.mdx:1361]: Great points all around. Let me do another rev where have both options listed, but do a little progressive disclosure to link out to our github docs so things don't get out of sync like Taylor mentions in his other comment.

File: src/pages/guidelines/2x-grid/implementation.mdx (hunk 2 above)
Comment (sstrubberg) [reply src/pages/guidelines/2x-grid/implementation.mdx]: Ok, I think I may have found a middle ground. Let me know what you think.

**Comment by aagonzales** [comment]
@sstrubberg Is this ready for review again?

**Comment by sstrubberg** [comment]
> @sstrubberg Is this ready for review again?

Yeppers!

**Review by tay1orjones — APPROVED** [review]

**Review by jnm2377 — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]

**Pull request merged**